src/client/utils/getExternalURI.ts

```diff
@@ -20,7 +19,7 @@ export function getExternalURI(documentPath: string, baseURI: string): string {
   const base = sanitizedBase.replace(END_SLASH_PATTERN, "");
   const route = sanitizedRoute.replace(START_SLASH_PATTERN, "");
 
-  const baseParts = base.replace(PROTOCOL_PATTERN, "").split("/");
+  const baseParts = base.split("/");
```

Comment:
I suggest doing a little smarter split/replace here so that we don't end up with `["https:", "", "www.example.com"]` here.

Rather we should to do something like:

```js
const [protocol, baseWithoutProtocol] = base.split(PROTOCOL_PATTERN);
const baseParts = baseWithoutProtocol.split('/');
```
and then add the protocol back at the end.

Also need to verify bases without protocols and see how to handle that

Reply:
Yeah, that wont work how i wrote it when there isnt a protocol, the array returned is a single item, the original base

Reply:
Do this before creating baseParts

```js
const [protocol] = PROTOCOL_PATTERN.exec(base) || [""];
```

And then at the end

```js
return `${protocol}${externalBase}/${routeParts.join("/")}`
```